sequence_prediction/sequence_generator/sequence_generator.py

```diff
@@ -0,0 +1,55 @@
+# ----------------------------------------------------------------------
+# Numenta Platform for Intelligent Computing (NuPIC)
+# Copyright (C) 2015, Numenta, Inc.  Unless you have an agreement
+# with Numenta, Inc., for a separate license for this software code, the
+# following terms and conditions apply:
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+# See the GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see http://www.gnu.org/licenses.
+#
+# http://numenta.org/licenses/
+# ----------------------------------------------------------------------
+
+import random
+
+
+
+class SequenceGenerator(object):
+
+  def generate(self, order, numPredictions=1):
+    assert order > 0
+
+    symbols = range(order + 1 + (numPredictions * 2))
+    random.shuffle(symbols)
+    subsequence = symbols[0:order-1]
+
+    sequences = []
+
+    for i in xrange(2):
+      start = order+i-1
+
+      for j in xrange(numPredictions):
+        end = -((numPredictions * i) + j + 1)
```

Comment:
The end symbol of each sequence is different. This is problematic because it is no longer necessary to use the start symbol to disambiguate the sequence.